arch/arm/core/aarch32/cortex_m/fault.c

```diff
@@ -929,52 +932,48 @@ static inline z_arch_esf_t *get_esf(uint32_t msp, uint32_t psp, uint32_t exc_ret
  * @param msp MSP value immediately after the exception occurred
  * @param psp PSP value immediately after the exception occurred
  * @param exc_return EXC_RETURN value present in LR after exception entry.
- *
+ * @param callee_regs Registers that are not part of the hardware esf (r4-r11)
  */
-void z_arm_fault(uint32_t msp, uint32_t psp, uint32_t exc_return)
+void z_arm_fault(uint32_t msp, uint32_t psp, uint32_t exc_return
```

Comment:
Instead, could you keep the same signature in all cases and then just "not use" the last argument #if !defined(CONFIG_EXTRA_EXCEPTION_INFO)??